app/src/ui/app.tsx

```diff
@@ -1415,9 +1419,10 @@ export class App extends React.Component<IAppProps, IAppState> {
       this.state.currentFoldout &&
       this.state.currentFoldout.type === FoldoutType.AppMenu
 
-    // As Linux still uses the classic Electron menu, we are opting out of the
-    // custom menu that is shown as part of the title bar below
-    if (__LINUX__) {
+    if (
+      __LINUX__ &&
+      this.props.appStore.getState().titleBarStyle === 'native'
```

Comment:
:art: this is also available directly

```suggestion
      this.state.titleBarStyle === 'native'
```